fv3core/utils/global_config.py

```diff
@@ -66,20 +55,30 @@ def is_gtc_backend() -> bool:
     return get_backend().startswith("gtc")
 
 
+def read_backend_options_file():
+    file = resources.open_binary("fv3core", "gt4py_options.yml")
+    if file:
+        options = yaml.safe_load(file)
+        file.close()
+        return options
+    raise FileNotFoundError("config file 'fv3core/gt4py_options.yml' not found")
+
+
 class StencilConfig(Hashable):
+    _all_backend_opts: Optional[Dict[str, Any]] = None
+
     def __init__(
         self,
         backend: str,
         rebuild: bool,
         validate_args: bool,
-        format_source: bool,
-        device_sync: bool,
+        format_source: Optional[bool] = None,
+        device_sync: Optional[bool] = None,
     ):
         self.backend = backend
         self.rebuild = rebuild
         self.validate_args = validate_args
-        self.format_source = format_source
-        self.device_sync = device_sync
+        self.backend_opts = self._get_backend_opts(device_sync, format_source)
```

Comment:
`verbose` option?

Reply:
The `verbose` option is controlled through the YAML file, `device_sync` and `format_source` are included here to maintain backward compatibility with the existing tests. I can happily add `verbose` here too if you like, however.